homeassistant/components/crownstone/__init__.py

```diff
@@ -0,0 +1,26 @@
+"""Integration for Crownstone."""
+from __future__ import annotations
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+from .entry_manager import CrownstoneEntryManager
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Initiate setup for a Crownstone config entry."""
+    manager = CrownstoneEntryManager(hass, entry)
+
+    if not await manager.async_setup():
+        return False
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok: bool = await hass.data[DOMAIN][entry.entry_id].async_unload()
+    if len(hass.data[DOMAIN]) == 0:
```

Comment:
This will never be 0 unless you pop the `entry.entry_id` as well after the successful unload.

Reply:
The pop happens in `entry_manager.py` , line 177, in `async_unload()`. The unload code was based on that in the Hue integration